pypeit/spectrographs/keck_lris.py

```diff
@@ -136,20 +136,40 @@ def init_meta(self):
         # Ingest
         self.meta = meta
 
+    # TODO Compound meta should have an ignore_bad_headers which should be True by default. Same goes for other meta data.
     def compound_meta(self, headarr, meta_key):
         if meta_key == 'binning':
             binspatial, binspec = parse.parse_binning(headarr[0]['BINNING'])
             binning = parse.binning2string(binspec, binspatial)
             return binning
         elif 'lampstat' in meta_key:
             idx = int(meta_key[-2:])
-            curr_date = time.Time(headarr[0]['MJD-OBS'], format='mjd')
+            # JFH: Since the golden rule is pypeit_setup should NEVER CRASH, even if headers are corrupt, we throw
+            # warnings and attempt to determine the date as best we can. If not a bogus MJD is used.
+            # TODO Which BOGUS MJD should we use??
+            try:
```

Comment:
Am not keen on a series of try/except loops.

Reply:
Please feel free to suggest a more elegant solution, or code it up yourself. I'm happy with anything that stops crashes for bad headers.

Reply:
The routine is flawed because it presumes all header cards exist, which they won't if headers are corrupted. Try except is designed for this kind of situation. Anyway, you either need an if else or a try except.

Reply:
We have handled this with VLT/X-Shooter.
How is this different?

Reply:
One layer of try except is removed. I'm not sure I follow the point about VLT-XSHOOTER? Can you be more specific or point to code.

Reply:
@freddavies can you run this branch on your data and see if it is fixed.

Reply:
See the `required_ftypes` in `init_meta()` of `vlt_xshooter.py`